lib/RequestRouter.js

```diff
@@ -336,7 +337,7 @@ function validateAndCastObject(object, schema, parentKey) {
             for (var arrayKey in value) {
                 arrayValue = validateAndCastParam.call(this, key, value[arrayKey], schemaProperty.items);
                 if (arrayValue && arrayValue.errorCode) {
-                    this.addPropertyError(parentKey + key, arrayValue.errorCode);
+                    this.addPropertyError(parentKey + key + '.' + arrayKey, arrayValue.errorCode);
```

Comment:
the property error should be in the monocle-props format, e.g. `myObj.myProperty@myPluckedKey`